mmdet/datasets/wider_face.py

```diff
@@ -15,40 +17,72 @@ class WIDERFaceDataset(XMLDataset):
     Conversion scripts can be found in
     https://github.com/sovrasov/wider-face-pascal-voc-annotations
     """
-    CLASSES = ('face', )
+    METAINFO = {'CLASSES': ('face', ), 'PALETTE': [(0, 255, 0)]}
 
-    PALETTE = [(0, 255, 0)]
+    def load_data_list(self) -> List[dict]:
+        """Load annotation from XML style ann_file.
 
-    def __init__(self, **kwargs):
-        super(WIDERFaceDataset, self).__init__(**kwargs)
+        Returns:
+            list[dict]: Annotation info from XML file.
+        """
+        assert self._metainfo.get('CLASSES', None) is not None, \
+            'CLASSES in `XMLDataset` can not be None.'
+        self.cat2label = {
+            cat: i
+            for i, cat in enumerate(self._metainfo['CLASSES'])
+        }
+
+        data_list = []
+        img_ids = mmcv.list_from_file(
```

Comment:
I remember `mmengine` also have `list_from_file` maybe you can change to use mmengine